activestorage/lib/active_storage/service.rb

```diff
@@ -60,10 +61,18 @@ def configure(service_name, configurations)
       #
       # See MirrorService for an example.
       def build(configurator:, name:, service: nil, **service_config) # :nodoc:
+        if service_config.key?(:uri)
```

Comment:
Why not have `uri` as a kwarg? Think that might clean this up rather than you're overwriting the hash that you got the key from.